maimai/src/js/maimai.table-config.js

```diff
@@ -466,9 +493,16 @@ $(document).ready(function() {
       {
         displayTitle: "BPM",
         name: "bpm",
-        data: "bpm",
+        data: function(row) {
+          return extractBPM(row.bpm);
+        },
         defaultContent: "",
         className: "details bpm",
+        customDropdownSortSource: function (row_a, row_b) {
+          var a = extractBPM(row_a.bpm).padStart(3, '0');
+          var b = extractBPM(row_b.bpm).padStart(3, '0');
+          return a.localeCompare(b);
+        },
```

Comment:
Suggest keeping original data as is, use as is for 'display' type (the actual printed value on table & modal) but use the extracted BPM for sort

So that we can still see the BPM range instead of flattening it for all

```suggestion
        data: "bpm",
        defaultContent: "",
        className: "details bpm",
        customDropdownSortSource: function (row_a, row_b) {
          var a = extractBPM(row_a.bpm).padStart(3, '0');
          var b = extractBPM(row_b.bpm).padStart(3, '0');
          return a.localeCompare(b);
        },
        render: function ( data, type, row ) {
          if ( type === 'display' && data ) {
            return data;
          }
          else {
            return extractBPM(row.bpm);
          }
        },
```